### PR TITLE
Fix dual packages ES module import bug

### LINKED ISSUE #904: @linaria/esbuild bundler shows error "TypeError: linaria is not a function"
## Environment

- Linaria version: 3.0.0-beta.15
- Bundler (+ version): @linaria/esbuild 3.0.0-beta.15 with esbuild v0.14.11
- Node.js version: v16.13.1
- OS: POP-OS (Ubuntu)

## Description

I followed your instructions on https://github.com/callstack/linaria/blob/master/docs/BUNDLERS_INTEGRATION.md#esbuild to add the `@linaria/esbuild` plugin to esbuild. However, when I add the import as follows it returns an object (containing the `default` property) instead of the plugin function itself.

```
// My esbuild.mjs module
import {build} from 'esbuild'
import linaria from '@linaria/esbuild'

  build({
    ...
    plugins: [linaria()],
  }).then(() => {
```

The above logic produces the error, `TypeError: linaria is not a function`. I can get around this issue by coding the plugins as follows:
```
import linaria from '@linaria/esbuild'

  build({
    ...
    plugins: [linaria.default()],
  }).then(() => {
```

### PR DESCRIPTION
**Pull request by iMoses**
Fixes #904 (and possibly #1043)

Solution is based on this documentation: https://nodejs.org/api/packages.html#approach-1-use-an-es-module-wrapper

It properly configures the package entry point for both CommonJS and ESM, which is currently not the case (see #904)

### CODE REVIEW

**Comment by iMoses** [comment]
@Anber seems that the problem is mainly with the integration packages (`@linaria/rollup`, `@linaria/esbuild`) and only possibly with `@linaria/react`

you wanna try applying the fix only there and see if it helps the ci processes?

File: packages/react/package.json [hunk 1]
```diff
@@ -53,8 +53,11 @@
       "styled": "./lib/processors/styled.js"
     }
   },
-  "main": "lib/index.js",
-  "module": "esm/index.js",
+  "type": "module",
+  "exports": {
```
Comment (Anber) [inline packages/react/package.json:57]: This file already has `exports`

**Comment by iMoses** [comment]
i'll try to fix this issues locally first

**Comment by cojoclaudiu** [comment]
@iMoses, @Anber this will be the fix for 
```
error when starting dev server:
TypeError: linaria is not a function
```
?

**Comment by cojoclaudiu** [comment]
@Anber I know the example but it's not working. The verisions of the vite and linaria in that example are not the latest. With the latest verision is not working.

@iMoses I will have a look locally as well and I will be back.

**Comment by cojoclaudiu** [comment]
> @cojoclaudiu yes, it should be
> 
> currently I only tested it locally with `@linaria/rollup` on Node v16 and it did the trick

How did you approuch the change for react/package.json since there it is an export already?

**Comment by iMoses** [comment]
@cojoclaudiu I haven't, I'll need to debug it later today after I get the environment ready to run the tests locally

**Comment by cojoclaudiu** [comment]
> @cojoclaudiu I haven't, I'll need to debug it later today after I get the environment ready to run the tests locally

I followed you last commit https://github.com/callstack/linaria/pull/1057/commits/24845e8e07db2da5e8c8e6c7af0bf1d205e5ae0f and updated locally react/package.json and I get this back

```
src/components/Button/Button.tsx: require() of ES Module /node_modules/@linaria/react/lib/processors/styled.js from /node_modules/@linaria/babel-preset/lib/utils/getTagProcessor.js not supported.
styled.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename styled.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /node_modules/@linaria/react/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

  14 | }
  15 | var _jsxFileName = "/src/components/Button/Button.tsx";
> 16 | import { styled } from "@linaria/react";
     |          ^^^^^^
  17 | import { jsxDEV as _jsxDEV } from "react/jsx-dev-runtime";
  18 | const ButtonContainer = styled.div`
  19 |  margin-top: 40px;
16:33:26 [vite] Internal server error: /src/components/Button/Button.tsx: require() of ES Module /Users/claudiu.cojocaru/node_modules/@linaria/react/lib/processors/styled.js from /node_modules/@linaria/babel-preset/lib/utils/getTagProcessor.js not supported.
styled.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename styled.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /node_modules/@linaria/react/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

  14 | }
  15 | var _jsxFileName = "/src/components/Button/Button.tsx";
> 16 | import { styled } from "@linaria/react";
     |          ^^^^^^
  17 | import { jsxDEV as _jsxDEV } from "react/jsx-dev-runtime";
  18 | const ButtonContainer = styled.div`
  19 |  margin-top: 40px;
  Plugin: linaria

**Comment by Anber** [comment]
@cojoclaudiu @iMoses I have fixed a couple of bugs with rollup@3 and the Vite example works well now. Could you please check this issue again?

**Comment by cojoclaudiu** [comment]
> @cojoclaudiu @iMoses I have fixed a couple of bugs with rollup@3 and the Vite example works well now. Could you please check this issue again?

I'm going to check now.

**Comment by Anber** [comment]
@cojoclaudiu okay… I'll take a look later today.

**Comment by cojoclaudiu** [comment]
> @cojoclaudiu okay… I'll take a look later today.

let me know later, maybe I can dig more as well after work

**Comment by Anber** [comment]
Looks like we just need to rename all files in esm folders from `js` to `mjs`. Tomorrow I'll try to change the build system in order to support `mjs`.

**Review by Anber — APPROVED** [review]

**Comment by Anber** [comment]
@all-contributors please add @iMoses for code

**Pull request merged**